int32 state # state: 0-idle 1-doing 2-Done
bool is_done
int32 errorcode